test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -331,3 +331,30 @@
    (with-puppet-conf puppet-conf-file conf-dir))
   ([puppet-conf-file dest-dir]
    (with-puppet-conf-fixture {"puppet.conf" puppet-conf-file} dest-dir)))
+
+(defn copy-config-files
+  [dirs-to-copy]
+  (doseq [[src target] dirs-to-copy]
+    (fs/mkdirs target)
+    (let [[_ dirs files] (first (fs/iterate-dir src))]
+      (doseq [d dirs] (fs/copy-dir (fs/file src d) target))
+      (doseq [f files] (fs/copy (fs/file src f) (fs/file target f))))))
+
+(defn remove-config-files
+  [dirs-to-copy]
+  (doseq [[src target] dirs-to-copy]
+    (let [[_ dirs files] (first (fs/iterate-dir src))]
+      (doseq [d dirs] (fs/delete-dir (fs/file target d)))
+      (doseq [f files] (fs/delete (fs/file target f))))))
+
+(defmacro with-puppet-config-files
```

Comment:
@rlinehan I agree, the differerences in names here are confusing.

I think the biggest difference between the two is that `with-puppet-conf-files` allows you to provide a list of files (potentially from different source directories) that you want to have copied over to a target directory, like:

`{"puppet.conf" "/my/source/dir1",
   "puppetdb.conf" "/my/source/dir2"} "/my/target/dir"`

... whereas `with-puppet-config-files` (which I just lifted verbatim from PE Puppet Server along with the tests which used it) allows you to copy multiple source directories to corresponding target directories, like:

`{"/my/source/dir1" "/my/target/dir1",
   "/my/source/dir2" "/my/target/dir2"}`

I suppose both capabilities might be useful.  For the new tests being imported, the ability to have multiple target directories - confdir and codedir - is a useful thing that you can't get out of `with-puppet-conf-files`.

Would just a name change for now make it a little better - like calling the new function this PR would bring in `with-puppet-config-dirs` - or even just `with-config-dirs` since there isn't anything specifically "Puppet"-specific about them?